system/default_ad_api/src/routing.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "routing.hpp"
+
+namespace default_ad_api
+{
+
+RoutingNode::RoutingNode(const rclcpp::NodeOptions & options) : Node("routing", options)
+{
+  const auto node = component_interface_utils::NodeAdaptor(this);
+  group_srv_ = create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  node.relay_message(pub_route_state_, sub_route_state_);
+  node.relay_message(pub_route_, sub_route_);
```

Comment:
[Low-priority] By the way, wouldn't it be better to switch the order of the args to be the same order as Linux commands such as `cp`?

```cpp
// Before
relay_message("/api/foo", "/planning/bar/baz");

// Before
relay_message("/planning/bar/baz", "/api/foo");
```

Reply:
I think `cp` command order is source and then destination. So the publisher (data source) is left.

Reply:
So, do you mean the API topic is the source and copy it to the Planning namespace? I didn't recognize so.

Reply:
Ah, you're right. I misunderstood.

Reply:
But it's not that important and there are other PR dependencies, so I'll keep it and replace later if necessary.

Reply:
Okay, thanks!